exercicios/para-casa/marina_oliveira_sistema_de_alunas.py

```diff
@@ -0,0 +1,141 @@
+
+def obter_opcao():
+    codigo_opcao = 0
+
+    while codigo_opcao not in [1, 2, 3, 4, 5, 6]:
+        try:
+            codigo_opcao = int(input("\nEscolha uma opção:\n"
+                                    "1 - Incluir uma nova aluna\n"
+                                    "2 - Consultar lista de alunas\n"
+                                    "3 - Consultar faltas da aluna\n"
+                                    "4 - Consultar notas da aluna\n"
+                                    "5 - Consultar status de aprovação\n"
+                                    "6 - Sair do sistema\n"
+                                    "Opção: "))
+                
+            if codigo_opcao not in [1, 2, 3, 4, 5, 6]:
+                print("Opção inválida. Por favor, escolha uma opção válida (1 a 5).\n")
+        except ValueError:
+            print("Entrada inválida. Por favor, digite um número inteiro.\n")
+            
+        return codigo_opcao
+    
+def incluir_nova_aluna():
+    print("\nInsira os dados da aula! ")
+    nome = input("Nome da aluna:")
+    sobrenome = input("Sobrenome da aluna: ")
+    turma = input("Turma da aluna: ")
+    notas = obter_nota()
+    presenca = obter_presenca()
+    participacao = float(input("Nota de participação da aluna: "))
+
+    salvar_dados_alunas(nome, sobrenome, turma, notas, presenca, participacao)
+
+    return nome
+
+def obter_nota():
+    quantidade_notas = input("Quantas notas deseja inserir? ")
+    notas = []
+
+    for i in range(int(quantidade_notas)):
+        while True:
+            entrada = input(f"Insira a nota {i + 1}: ")   
+            try: 
+                nota = float(entrada)
+                notas.append(nota)
+                break
+            except ValueError:
+                print("Entrada inválida. Por favor, insira true ou false.")
+
+    return notas 
+
+def obter_presenca(): 
+    quantidade_aulas = input("Insira a quantidade de aulas da aluna: ")
+    presenca = []
+
+    for i in range(int(quantidade_aulas)):
+        while True:
+            entrada = input(f"Presença na aula {i + 1} (True/False): ")
+            if entrada.lower() == 'true':
+                presenca.append(True)
+                break
+            elif entrada.lower() == 'false':
+                presenca.append(False)
+                break
+            else:
+                print("Entrada inválida. Por favor, digite 'True' ou 'False'.")
+
+    return presenca    
+
+def salvar_dados_alunas(nome, sobrenome, turma, notas, presenca, participacao):
+    chave = (nome, sobrenome)
+    dataset[chave] = {
+        "Turma": turma,
+        "Notas": notas,
+        "Presença": presenca,
+        "Participação": participacao,
+    }
+    
+    print(f"Aluna {nome} {sobrenome} adicionada com sucesso!")
+    
+    
+def consultar_lista_alunas():
+    if dataset:
+        print("\nAlunas cadastradas:")
+        for nome in dataset:
+            print(nome)
```

Comment:
No requisito funcional, era esperado que a impressão dos nomes das alunas estivesse formatado. Do modo como está, ele está imprimindo a tupla de forma crua.

**Sugestão:**
```suggestion
    for nome, sobrenome in dataset:
        print(f"{nome} {sobrenome}")
```